src/test/scala/chisel3/experimental/dataview/ReifySpec.scala

```diff
@@ -0,0 +1,318 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.experimental.dataview
+
+import circt.stage.ChiselStage
+import chisel3._
+import chisel3.probe.Probe
+import chisel3.properties.Property
+import chisel3.experimental.Analog
+import chisel3.experimental.dataview._
+
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers._
+import chisel3.reflect.DataMirror
+import chisel3.experimental.hierarchy.{instantiable, public, Instantiate}
+
+object ReifySpec {
+  // Views can be 1-1 but still not identity
+  type ReversedVec[T <: Data] = Vec[T]
+  implicit def reversedVecView[T <: Data]: DataView[Vec[T], ReversedVec[T]] =
+    DataView.mapping[Vec[T], ReversedVec[T]](v => v.cloneType, { case (a, b) => a.reverse.zip(b) })
+
+  class AllElementsBundle extends Bundle {
+    val u = UInt(8.W)
+    val s = SInt(8.W)
+    val b = Bool()
+    val r = Reset()
+    val d = AsyncReset()
+    val c = Clock()
+    val a = Analog(8.W)
+    val p = Probe(UInt(8.W))
+    val prop = Property[String]()
+  }
+  implicit val allElementsView: DataView[Seq[Data], AllElementsBundle] =
+    DataView.mapping[Seq[Data], AllElementsBundle](
+      _ => new AllElementsBundle,
+      { case (a, b) => a.zip(b.getElements) }
+    )
+
+  class SimpleBundle extends Bundle {
+    val value = UInt(8.W)
+  }
+  class NestedBundle extends Bundle {
+    val child = new SimpleBundle
+  }
+  class TargetBundle extends Bundle {
+    val fizz = new NestedBundle
+    val buzz = new SimpleBundle
+    val vec = Vec(2, UInt(8.W))
+  }
+  class ViewChildBundle extends Bundle {
+    val a = Vec(2, UInt(8.W))
+    val b = Vec(2, UInt(8.W))
+  }
+  class ViewBundle extends Bundle {
+    val foo = UInt(8.W) // used to ensure ViewBundle isn't 1-1
+    val bar = new ViewChildBundle
+  }
+  // The key thing about this mapping is that ViewBundle.bar maps 1-1 to TargetBundle
+  // but the relationship is tricky.
+  // TODO should we add an aggregate child of ViewBundle that maps 1-1 to an aggregate child of TargetBundle?
+  implicit val myView: DataView[(UInt, TargetBundle), ViewBundle] =
+    DataView.mapping[(UInt, TargetBundle), ViewBundle](
+      _ => new ViewBundle,
+      {
+        case ((u, t), v) =>
+          Seq(u -> v.foo, t.fizz.child.value -> v.bar.a(0), t.buzz.value -> v.bar.a(1), t.vec -> v.bar.b)
+      }
+    )
+}
+
+import ReifySpec._
+
+class ReifySpec extends AnyFunSpec {
+
+  describe("dataview.reify") {
+
+    it("should reify single targets and identity for all non-view Elements") {
+      ChiselStage.emitCHIRRTL(new Module {
```

Comment:
Slightly faster to only run. Ditto throughout:

```suggestion
      ChiselStage.elaborate(new Module {
```